scripts/o3de/o3de/repo_properties.py

```diff
@@ -293,6 +293,44 @@ def _auto_update_json(object_type: str or list,
             _edit_objects(object_name, validation.valid_o3de_template_json, repo_json, expected_files.get("template.json"))
     return 0
 
+def _dry_run(repo_json: dict,
```

Comment:
Suggest changing this method name to "print_repo_diff". The name "dry_run" seems like this is different logic to dry run, but really this is just pretty printing the differences between 2 repo files.

Reply:
Or use python's [difflib](https://docs.python.org/3/library/difflib.html) as @spham-amzn suggested. Not sure how hard it would be to "pretty-print" the differences like you are here. There's a lot of noise in repo.json files since the all the object (gem/template/project) properties are added into the repo.json, not something nice and streamline looking such as just the object name (which you're printing now)